src/vehicle.cpp

```diff
@@ -6298,23 +6298,20 @@ bool vehicle::fire_turret( int p, bool manual )
     
     // Create a fake gun
     // @todo damage the gun based on part hp
-    item gun( turret_data.gun.typeId(), turret_data.gun.bday, charges );
-    gun.set_curammo( turret_data.gun.ammo_current() );
-    gun.update_charger_gun_ammo();
-
-    long charges_left = charges;
+    item gun( turret_data.gun.typeId(), turret_data.gun.bday );
+    if( gun.is_charger_gun() ) {
+        gun.update_charger_gun_ammo();
+    } else if( turret_data.gun.ammo_current() != "null" ) {
+        gun.ammo_set( turret_data.gun.ammo_current(), charges );
+    }
 
     // TODO sometime: change that g->u to a parameter, so that NPCs can shoot too
-    const bool success = manual ?
     // TODO: unify those two functions.
-        manual_fire_turret( p, g->u, gun, charges_left ) :
-        automatic_fire_turret( p, gun, charges_left );
-    if( success ) {
-        turret_data.consume( *this, p, charges - charges_left );
-    }
+    int shots = manual ? manual_fire_turret( p, g->u, gun ) : automatic_fire_turret( p, gun );
+    turret_data.consume( *this, p, shots * gun.ammo_required() );
 
     // If manual, we need to know if the shot was actually executed
-    return !manual || success;
+    return shots > 0 || manual == false;
```

Comment:
Comparing a `bool` (`manual`) to a constant expression is redundant - that's what negation is for.